cmake/modules/extensions.cmake

```
@@ -718,6 +718,222 @@ function(generate_inc_file_for_target
+# named optional parameters.
+#
+# These functions provide a higher-level embedding mechanism compared to
+# generate_inc_* functions, enabling advanced use cases like:
+#
+#   - Embedding of data from more than one files, where the list and number
+#     of files being included is determined at build time, e.g. driven by
+#     some device-tree fields or config options. generate_inc_* functions
+#     cannot handle such use-cases without a fair amount of additional custom
+#     support code added to corresponding CMake build files.
+#
+#   - generate_inc_* functions embed binary data by first converting it
+#     into C-source compilable form; which is fine for small binary files
+#     but for larger files this can increase build time and resources usage
+#     significantly. Whereas these new functions can transparently make use
+#     of GNU-style "incbin" directive, when available (and usable), for much
+#     faster inclusion of binary files, automatically falling back to using
+#     generate_inc_* functions internally, for other cases.
+#
+# Required named parameters:
+#
+#   OUTPUT        - The generated file.
+#   BINARY_FILES  - List of binary files to be converted into embeddable form.
+#   SYMBOL_NAMES  - List of symbol names corresponding to binary files being embedded.
+#
+# Optional named parameters:
+#
+#   OFFSET        - Offset, in bytes, from which data inclusion from the
+#                   specified files should be started.
+#   LENGTH        - Length, in bytes, of the data subset to be included from
+#                   the specified files.
+#   READONLY      - yes (default) or no.
+#   ALIGN         - Alignment for data being embedded in bytes. Default is 4.
+#   SECTION       - Name of the desired linker section. Data from all specified
+#                   binary files will be placed in the given section name.
+#   COMPRESS      - Compress the data with specified compression method: gzip or no (default).
+#   GZIP_MTIME    - mtime seconds in the gzip header.
+#                   Defaults to zero to keep builds deterministic. For
+#                   current date and time, set this option to "now".
+#
+# Examples:
+#
+#   generate_binary_embed_inc_file_for_target(app
+#     OUTPUT        ${gen_dir}/test_data_embed_combined.inc
+#     BINARY_FILES  test1.bin test2.bin
+#     SYMBOL_NAMES  symbol_names test1_data test2_data
+#     ALIGN         16
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/test_data_embed.inc
+#     BINARY_FILES  test_data.bin
+#     SYMBOL_NAMES  test_data
+#     OFFSET        5
+#     LENGTH        52
+#     SECTION       tdata
+#     READONLY      no
+#     ALIGN         32
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/raw_data.inc
+#     BINARY_FILES  ${EXTERNAL_SDK_PATH}/testx_file.bin
+#     SYMBOL_NAMES  test_data
+#   )
+#
+function(generate_binary_embed_inc_file_for_target target)
```

Just a single function, target can be an optional argument which defaults to `ZEPHYR_CURRENT_LIBRARY`.
```suggestion
function(generate_binary_embed_inc_file)
```
like this: `generate_binary_embed_inc_file([TARGET <target>] ...)`

and then remove this: https://github.com/zephyrproject-rtos/zephyr/blob/e1726295de1905cdd2c9fc2e3112710d5fa449bd/cmake/modules/extensions.cmake#L933